checker/tests/index/ArrayWrapper.java

```diff
@@ -0,0 +1,47 @@
+// Test case for https://github.com/kelloggm/checker-framework/issues/154
+// This class wraps an array, but doesn't expose the array in its public interface. This test
+// ensures that indexes for this new collection can be annotated as if the collection were an array.
+
+import org.checkerframework.checker.index.qual.IndexFor;
+import org.checkerframework.checker.index.qual.LengthOf;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.index.qual.SameLen;
+
+/** ArrayWrapper is a fixed-size, weakly-typed generic collection. */
```

Comment:
In what sense is this weakly-typed?
The abstraction looks strongly-typed to me.  (The implementation uses a cast, but that doesn't jeopardize type safety.)
This remark also appears in the manual section; keep it in sync with this code.

Reply:
I think this was just a mistake in my memory. For some reason, I remembered Java collection classes being classified as either "strongly" typed (backed by a generic array - so `internalRep` would be a `T[]`) or "weakly" typed (backed by an array of `Object`s). I can't corroborate this distinction with some quick googling, so I'll just remove the reference (it's not really important anyway).